pennylane/vqe/vqe.py

```diff
@@ -96,14 +129,38 @@ class Hamiltonian(qml.operation.Observable):
     >>> H3 = qml.Hamiltonian(torch.tensor([1., 2., 3.]), [qml.PauliX(0), qml.PauliY(0), qml.PauliX(1)])
     >>> H3.compare(H1 + H2)
     True
+
+    A Hamiltonian can store information on which commuting observables should be measured together in
+    a circuit:
+
+    >>> obs = [qml.PauliX(0), qml.PauliX(1), qml.PauliZ(0)]
+    >>> coeffs = np.array([1., 2., 3.])
+    >>> H = qml.Hamiltonian(coeffs, obs, compute_grouping=True)
+    >>> H.grouping_indices
+    [[0, 1], [2]]
```

Comment:
I thought they are computed when the tapes for each group are evaluated?

Reply:
Oh wait... that's a good point. I think I am confused, because the grouping module duplicates this 🤔 

https://github.com/PennyLaneAI/pennylane/blob/5c0adbc0960d65094708bd41b6ce9d0ec514496f/pennylane/grouping/optimize_measurements.py#L61

Reply:
I was never sure if it was duplication or something unique

Reply:
We should probably remove this duplication overhead at some point 😆

Reply:
Yes!!